examples/structured_data/time_series_anomaly_using_autoencoder.py

```diff
@@ -0,0 +1,344 @@
+"""
+Title: Timeseries anomaly detection using an Autoencoder
+Author: [pavithrasv](https://github.com/pavithrasv)
+Date created: 2020/05/31
+Last modified: 2020/05/31
+Description: Detect anomalies in a timeseries using an Autoencoder.
+"""
+
+"""
+## Introduction
+
+This script demonstrates how you can use a reconstruction convolutional autoencoder model
+to detect anomolies in timeseries data.
+
+We train the model on timeseries data which does not have any anomalies. We will use the
+[Numenta Anomaly Benchmark](https://www.kaggle.com/boltzmannbrain/nab?) dataset to
+demonstrate this.
+"""
+
+"""
+## Setup
+"""
+
+import io
+import matplotlib
+import numpy as np
+import os
+import pandas as pd
+import tensorflow as tf
+
+from datetime import datetime
+from matplotlib import pyplot as plt
+from matplotlib import dates as md
+from numpy.random import seed
+from tensorflow.keras import layers
+from tensorflow.keras import Sequential
+
+"""
+## Set random seed
+"""
+
+RANDOM_SEED = 2020
+seed(RANDOM_SEED)
+tf.random.set_seed(RANDOM_SEED)
+
+"""
+## Load the data
+
+We will use the  dataset [Numenta Anomaly Benchmark
+(NAB)](https://www.kaggle.com/boltzmannbrain/nab?) dataset. We will use the
+`art_daily_small_noise.csv` file for training and the `art_daily_jumpsup.csv` file for
+testing anomaly detection.
```

Comment:
Likewise, we should provide more context. Read your example from the perspective of a first-time reader.

Reply:
Done.